run-checks

```diff
@@ -0,0 +1,160 @@
+#!/bin/bash -e
+
+# shellcheck disable=SC2145
+
+#----------------------------------------------------#
+# PRINTING ROUTINES                                  #
+#----------------------------------------------------#
+
+P_RESET="\e[0m"
+P_RED="\e[31m"
+P_GREEN="\e[32m"
+
+print_newline() {
+	echo -e "$P_RESET"
+}
+
+print_normal() {
+	echo -e "$P_RESET$@"
+}
+
+print_red() {
+	echo -e "$P_RED$@$P_RESET"
+}
+
+print_green() {
+	echo -e "$P_GREEN$@$P_RESET"
+}
+
+print_special() {
+	echo -ne "$P_RESET$@"
+}
+
+print_err() {
+	print_newline
+	pr "Error: $1"
+	print_newline
+	exit 1
+}
+
+#----------------------------------------------------#
+# TEST OUTPUT CONTROL                                #
+#----------------------------------------------------#
+
+print_test() {
+	print_special "$@"
+}
+
+run() {
+        if $VERBOSE; then
+                v=$(exec 2>&1 && set -x && set -- "$@")
+		echo
+                echo "#${v#*--}"
+		set +e
+                "$@"
+		RET=$?
+		set -e
+        else
+		set +e
+                "$@" >/dev/null 2>&1
+		RET=$?
+		set -e
+        fi
+}
+
+return_check() {
+	if [ $RET -ne 0 ]; then
+                print_newline
+                print_red "Error: $1 (run 'run-checks --verbose' for details)"
+                print_newline 
+                exit 1
+	else
+		print_green "Pass"
+	fi
+}
+
+#----------------------------------------------------#
+# CMDLINE OPTIONS                                    #
+#----------------------------------------------------#
+
+ARGC=$#
+ARGV=("$@")
+
+VERBOSE=false
+HELP=false
+
+# Clear args
+set --
+
+COUNT=0
+while [ $COUNT -lt $ARGC ]; do
+	STRIP=${ARGV[COUNT]}
+
+	#ID=0
+	LONGFORMAT=$(echo "$STRIP" | cut -b 1-2)
+	if [ "$LONGFORMAT" = "--" ]; then
+		STRIP=$(echo "${ARGV[COUNT]}" | cut -b 2-3)
+	else
+		#ID=$(echo "$STRIP" | cut -b 3-)
+		STRIP=$(echo "$STRIP" | cut -b 1-2)
+	fi
+	
+
+	if [ "$STRIP" = "-v" ]; then
+		VERBOSE=true
+	elif [ "$STRIP" = "-h" ]; then
+		HELP=true
+	else
+		print_newline
+		print_red "Error: Unknown argument"
+		COUNT=$ARGC
+		HELP=true
+	fi
+COUNT=$((COUNT+1))
+done
+
+if [ "$HELP" = "true" ]; then
+	print_normal "Usage:"
+	print_newline
+	print_normal "run-checks [-v] [-h]"
+	print_newline
+	print_normal "Help:"
+	print_newline
+	print_normal "-v (--verbose)      Verbose output for operations"
```

Comment:
Maybe a case for YAGNI? if the only option supported is --verbose, maybe we just make it default and not provide cmdline args?

Reply:
Done :) Thanks.